website/docs/r/cdn_frontdoor_custom_domain_association.html.markdown

```diff
@@ -0,0 +1,57 @@
+---
+subcategory: "CDN"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_cdn_frontdoor_custom_domain_association"
+description: |-
+  Manages the single apply/single destroy incompatibility between the CDN FrontDoor API and the Terraform dependency graph.
+---
+
+# azurerm_cdn_frontdoor_custom_domain_association
+
+Manages the single destroy incompatibility between the CDN FrontDoor API and the Terraform dependency graph generation. You should use this association resource to avoid receiving the `This resource is still associated with a route. Please delete the association with the route first before deleting this resource` error when issuing the Terraform `destroy` command.
+
+## Example Usage
+
+```hcl
+resource "azurerm_cdn_frontdoor_custom_domain_association" "example" {
+  cdn_frontdoor_custom_domain_id = azurerm_cdn_frontdoor_custom_domain.contoso.id
+  cdn_frontdoor_route_ids        = [azurerm_cdn_frontdoor_route.contoso.id, azurerm_cdn_frontdoor_route.fabrikam.id]
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `cdn_frontdoor_custom_domain_id` - (Required) The resource ID of the CDN FrontDoor Custom Domain that should be managed by the association resource. Changing this forces a new association resource to be created.
```

Comment:
we use `ID` rather than `Resource ID` because it's an ambiguous term between Azure and Terraform:

```suggestion
* `cdn_frontdoor_custom_domain_id` - (Required) The ID of the CDN FrontDoor Custom Domain that should be managed by the association resource. Changing this forces a new association resource to be created.
```